e2e/cloud-plugins-suite/azure-monitor.spec.ts

```diff
@@ -0,0 +1,310 @@
+import { load } from 'js-yaml';
+
+import { e2e } from '@grafana/e2e';
+
+import { selectors } from '../../public/app/plugins/datasource/grafana-azure-monitor-datasource/e2e/selectors';
+import {
+  AzureDataSourceJsonData,
+  AzureDataSourceSecureJsonData,
+  AzureQueryType,
+} from '../../public/app/plugins/datasource/grafana-azure-monitor-datasource/types';
+
+import EXAMPLE_DASHBOARD from './example-dashboards/azure-monitor.json';
+
+const provisioningPath = `../../provisioning/datasources/azmonitor-ds.yaml`;
+const e2eSelectors = e2e.getSelectors(selectors.components);
+
+type AzureMonitorConfig = {
+  secureJsonData: AzureDataSourceSecureJsonData;
+  jsonData: AzureDataSourceJsonData;
+};
+
+type AzureMonitorProvision = { datasources: AzureMonitorConfig[] };
+
+function provisionAzureMonitorDatasources(datasources: AzureMonitorProvision[]) {
+  const datasource = datasources[0].datasources[0];
+
+  e2e()
+    .intercept(/subscriptions/)
+    .as('subscriptions');
```

Comment:
This allows us to intercept the subscriptions API request so that we can wait for the response on line 42 before continuing. I was previously having issues with the response being flaky in response time.